sys-tune/toolbox.json

```diff
@@ -0,0 +1,5 @@
+{
+    "name"  : "sys-tune",
+    "tid"   : "4200000000000000",
+    "requires_reboot": true
```

Comment:
Is this still true now? As discussed, if the IPC init fails, you should be able to get it to work again by calling svcManageNamedPort with session count 0.

Reply:
I forgot to try this sorry

Reply:
in main.cpp i added:
```c++
// not found can happen if the service isn't started
// connection refused can happen is the service was terminated by pmshell
if (R_VALUE(rc) == KERNELRESULT(NotFound) || KERNELRESULT(ConnectionRefused)) {
```

and in ipc_server.c
```c
Result ipcServerInit(IpcServer* server, const char* name, u32 max_sessions)
{
    if(max_sessions < 1 || max_sessions > (MAX_WAIT_OBJECTS - 1))
    {
        return MAKERESULT(Module_Libnx, LibnxError_BadInput);
    }

    server->srvName = smEncodeName(name);
    server->max = max_sessions + 1;
    server->count = 0;

    Result rc = svcManageNamedPort(&server->handles[0], server->srvName.name, max_sessions);
    if (R_FAILED(rc))
    {
        rc = svcManageNamedPort(&server->handles[0], server->srvName.name, 0);
        if(R_SUCCEEDED(rc))
        {
            svcCloseHandle(server->handles[0]);
            rc = svcManageNamedPort(&server->handles[0], server->srvName.name, max_sessions);
        }
    }

    if(R_SUCCEEDED(rc))
    {
        server->count = 1;
    }
    return rc;
}
```

this works for me. is this acceptable?

Reply:
Pretty sure the closehandle isn't needed.
Check atmosphere source to be sure.